### PR TITLE
Static Dispatch Mediator

### PR DESCRIPTION
**Pull request by epicbrownie**
### Overview
I've created a second implementation of `IMediator` that has been named `StaticDispatchMediator`.  Rather than using `dynamic` to route requests to their respective handlers, "dispatcher" expression lambdas are created and cached at the time of configuration.  The old mediator has been renamed to `DynamicDispatchMediator`, but is still the default mediator to be registered to SimpleInjector when `container.ConfigureMediator(...)` is called.

### Breaking Changes
This code _should_ produce no breaking changes.

### Use
To use the new mediator, simply replace existing calls to `container.ConfigureMediator(...)` with `container.ConfigureStaticMediator(...)`.  This new configuration function bears the same signature, and the mediator behaves in the same manner as its dynamic counterpart.

### Benefits
It's roughly (and pretty consistently) about 430% faster than the dynamic mediator:

tm1

tm2

### CODE REVIEW

**Comment by justinobney** [comment]
I removed that integration.

File: Turner.Infrastructure.Mediator/Mediators/StaticDispatcher.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    {
+        private Dictionary<Type, Func<object, object>> _dispatchers;
+
+        public StaticDispatcher(Func<Type, object> handlerFactory, params Assembly[] assemblies)
+        {
+            Register(handlerFactory, assemblies);
+        }
+
+        public object Dispatch(object request) =>
+            _dispatchers[request.GetType()](request);
```
Comment (justinobney) [inline Turner.Infrastructure.Mediator/Mediators/StaticDispatcher.cs:19]: ❤️ this feels like JavaScript

File: Turner.Infrastructure.Mediator/Mediators/StaticDispatcher.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        {
+            _dispatchers = new Dictionary<Type, Func<object, object>>();
+
+            foreach (var assembly in assemblies)
+            {
+                var classTypes = assembly.GetTypes()
+                    .Where(type => type.IsClass && !type.IsAbstract && !type.IsInterface)
+                    .ToArray();
+
+                foreach (var type in classTypes
```
Comment (bcornett) [inline Turner.Infrastructure.Mediator/Mediators/StaticDispatcher.cs]: Can we pull the collection out to a variable as to make this more descriptive and readable?

Comment (bcornett) [reply Turner.Infrastructure.Mediator/Mediators/StaticDispatcher.cs]: Same with below

Comment (epicbrownie) [reply Turner.Infrastructure.Mediator/Mediators/StaticDispatcher.cs]: Indeed we can

**Review by bcornett — APPROVED** [review]

**Pull request merged**